SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -280,4 +281,183 @@ bool SpatialCommandUtils::HasDevLoginTag(const FString& DeploymentName, bool bIs
 	return false;
 }
 
+FProcHandle SpatialCommandUtils::StartLocalReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName, const FString& ListeningAddress, const int32 Port , FString &OutResult, int32 &OutExitCode)
+{
+	FString Command = FString::Printf(TEXT("cloud connect external %s --listening_address %s --local_receptionist_port %i"), *CloudDeploymentName, *ListeningAddress, Port);
+
+	if (bIsRunningInChina)
+	{
+		Command += SpatialGDKServicesConstants::ChinaEnvironmentArgument;
+	}
+
+	FProcHandle ProcHandle;
+
+	void* ReadPipe = nullptr;
+	void* WritePipe = nullptr;
+	ensure(FPlatformProcess::CreatePipe(ReadPipe, WritePipe));
+
+	ProcHandle = FPlatformProcess::CreateProc(*SpatialGDKServicesConstants::SpatialExe, *Command, false, true, true, nullptr, 1 /*PriorityModifer*/, *SpatialGDKServicesConstants::SpatialOSDirectory, WritePipe);
+
+	bool bProcessSucceeded = false;
+	bool bProcessFinished = false;
+	if (ProcHandle.IsValid())
+	{
+		while (!bProcessFinished && !bProcessSucceeded)
+		{
+			bProcessFinished = FPlatformProcess::GetProcReturnCode(ProcHandle, &OutExitCode);
+
+			OutResult = OutResult.Append(FPlatformProcess::ReadPipe(ReadPipe));
+			bProcessSucceeded = OutResult.Contains("The receptionist proxy is available");
+
+			FPlatformProcess::Sleep(0.01f);
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialCommandUtils, Error, TEXT("Execution failed. '%s' with arguments '%s' in directory '%s'"), *SpatialGDKServicesConstants::SpatialExe, *Command, *SpatialGDKServicesConstants::SpatialOSDirectory);
+	}
+
+	if (!bProcessSucceeded)
+	{
+		FPlatformProcess::TerminateProc(ProcHandle, true);
+		ProcHandle.Reset();
+	}
+
+	FPlatformProcess::ClosePipe(0, ReadPipe);
+	FPlatformProcess::ClosePipe(0, WritePipe);
+
+	return ProcHandle;
+}
+
+void SpatialCommandUtils::StopLocalReceptionistProxyServer(FProcHandle& ProcHandle)
+{
+	if (ProcHandle.IsValid())
+	{
+		FPlatformProcess::TerminateProc(ProcHandle, true);
+	}
+}
+
+bool SpatialCommandUtils::GetProcessName(const FString& PID, FString& OutProcessName)
+{
+
+#if PLATFORM_MAC
+	UE_LOG(LogSpatialCommandUtils, Warning, TEXT("Failed to get the name of the process that is blocking the required port. To get the name of the process in MacOS you need to use SpatialCommandUtils::GetProcessInfoFromPort."));
+	return false;
+#endif 
+
+	bool bSuccess = false;
```

Comment:
I believe this will cause `unreachable code` compiler warning. The windows part of this should be in an `#else` block.